File: src/backend/ClothingStoreManager/ClothingStoreManager/src/main/java/com/windowprogramming/ClothingStoreManager/dto/request/employee/EmployeeCreationRequest.java
package com.windowprogramming.ClothingStoreManager.dto.request.employee;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import lombok.AccessLevel;
import lombok.Builder;
import lombok.Getter;
import lombok.Setter;
import lombok.experimental.FieldDefaults;

import java.math.BigDecimal;
import java.time.LocalDate;

@Getter
@Setter
@Builder
@FieldDefaults(level = AccessLevel.PRIVATE)
public class EmployeeCreationRequest {
    @NotBlank(message = "REQUIRED_EMPLOYEE_NAME")
    String name;

    String phoneNumber;

    @NotBlank(message = "REQUIRED_EMPLOYEE_CITIZEN_ID")
    String citizenId;

    @NotBlank(message = "REQUIRED_EMPLOYEE_JOB_TITLE")
    String jobTitle;

    @NotNull(message = "REQUIRED_EMPLOYEE_SALARY")
    BigDecimal salary;
    String email;
    LocalDate dateOfBirth;
    String address;
    String area;
    String ward;
}

File: src/winui/Sale_Project.Core/Models/Employees/EmployeeCreationRequest.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Text.Json.Serialization;
using System.Threading.Tasks;

namespace Sale_Project.Core.Models.Employee;
public class EmployeeCreationRequest
{
    [JsonPropertyName("name")]
    public string Name
    {
        get; set;
    }

    [JsonPropertyName("phoneNumber")]
    public string PhoneNumber
    {
        get; set;
    }

    [JsonPropertyName("citizenId")]
    public string CitizenId
    {
        get; set;
    }

    [JsonPropertyName("jobTitle")]
    public string JobTitle
    {
        get; set;
    }

    [JsonPropertyName("salary")]
    public decimal Salary
    {
        get; set;
    }

    [JsonPropertyName("email")]
    public string Email
    {
        get; set;
    }

    [JsonPropertyName("dateOfBirth")]
    public DateTime DateOfBirth
    {
        get; set;
    }

    [JsonPropertyName("address")]
    public string Address
    {
        get; set;
    }

    [JsonPropertyName("area")]
    public string Area
    {
        get; set;
    }

    [JsonPropertyName("ward")]
    public string Ward
    {
        get; set;
    }
}
